test/coap_base_class_test.dart

```diff
@@ -57,14 +25,12 @@ void main() {
     });
 
     test('Parse wild card', () {
-      var res = CoapMediaType.parseWildcard(null);
-      expect(res, isNull);
-
-      res = CoapMediaType.parseWildcard('xml*');
-      expect(res, <int>[
+      final res = CoapMediaType.parseWildcard('xml*');
```

Comment:
Have only had a super brief look, but couldn't we have a `CoapMediaType.startsWith` instead, which is using the string method `startsWith`? It would perform better than a regex, and the code would be a bit simpler.

Reply:
Hmm, I think a `startsWith` method would not really catch the same semantics, as the current version of this method seems to also (or only?) find those Content-Formats that *end* with `xml`. Maybe we could also remove the method completely, though, since users could also use something like `CoapMediaType.values.where((value) => value.mimeType.endsWith('xml'))`.

Reply:
Oh, I thought `xml*` would match anything starting with `xml` and `*xml` would do the opposite.

Yeah, in general I think this method (and several others like it) may have made sense way back in .NET before linq/enumerables, but now in Dart it feels a bit superflous to re-implement this type of logic, `.where(...)` is providing a lot more flexibility with a small increase in verbosity. Using the Dart SDK also makes it clearer to the people using the lib what's going on.

Reply:
> Oh, I thought `xml*` would match anything starting with `xml` and `*xml` would do the opposite.

Hmm, yeah, you are right, that should be the case. I just got the impression that this is the current behavior, as all returned formats end with `xml`. Maybe it currently matches all Content-Formats that contain `xml`? The implementation does not seem that intuitive in general.

> Yeah, in general I think this method (and several others like it) may have made sense way back in .NET before linq/enumerables, but now in Dart it feels a bit superflous to re-implement this type of logic, .where(...) is providing a lot more flexibility with a small increase in verbosity. Using the Dart SDK also makes it clearer to the people using the lib what's going on.

That's true! Then I will clean up the file a bit more and remove the method :)

Reply:
Method is now removed :)